### ISSUE TITLE
Dashboards not loading after upgrading from v10.1.5 to v10.2.0

### ISSUE THREAD

**Issue by n8ores** [grafana/grafana#77082]
It seems that the public dashboards upgrade, may have broken the `ENV GF_AUTH_ANONYMOUS_ORG_ROLE "Viewer"` and `ENV GF_AUTH_ANONYMOUS_ENABLED="true"` attributes.

### What happened?

After upgrading from 10.1.5 to 10.2. we now see the following error for all our dashboards displayed in the logs after Grafana boots up. No dashboards will load and localhost:3000/dashboards simply gives a page not found 404. 

```
level=error msg="Could not make user admin" dashboard="<YourDashboardNameHere>" namespaceID=user userID=0 error="identifier is not initialized"_
```

When we revert back to 10.1.5 we instead see this error

```
level=warn msg="Failed to authenticate request" client=auth.client.session error="user token not found"
```

Note, this only happens when we enable anonymous logins (using the following env vars in the dockerfile)

ENV GF_AUTH_ANONYMOUS_ORG_ROLE "Viewer"

If we set that to ENV GF_AUTH_ANONYMOUS_ORG_ROLE "Admin" it fixes the problem, but we don't want users to be able to edit panels etc.

==========================================================

**grafana.ini**

```
[paths]
provisioning = /etc/grafana/provisioning

[server]
enable_gzip = true
# If you wish to host Grafana behind a SSL certificate, enable this.
# It is generally not required because we host Grafana within a kubernetes cluster
# and the HTTPS routing is already handled by the load balancer before hand.
# protocol = https
# cert_key = /etc/grafana/grafana.key
# cert_file = /etc/grafana/grafana.crt
root_url = /
serve_from_sub_path  = true
#router_logging = false	

[users]
default_theme = dark

[dashboards]
min_refresh_interval = 1s
```

**Dockerfile** 
Note: We are manually pre-downloading some plugins and injecting them into the container.

```
ARG GRAFANA_VERSION="latest"

# Stage 1: Conditional Copy for subset dashboard builds
FROM alpine as conditional-copy
ARG ENV_TYPE
COPY /dashboards/ /dashboards/
RUN mkdir -p /dashboard_subset && \
    if [ "$ENV_TYPE" = "45SLD" ]; then \
      mkdir -p /dashboard_subset/dashboards/; \
      cp /dashboards/space_weather.json /dashboard_subset/dashboards/; \
    else \
      cp -r /dashboards/ /dashboard_subset/; \
    fi

# Stage 2: Final build
FROM --platform=amd64 grafana/grafana:${GRAFANA_VERSION}

ARG GF_INSTALL_IMAGE_RENDERER_PLUGIN="false"

ARG GF_GID="0"

ENV GF_PATHS_PLUGINS="/var/lib/grafana-plugins"
ENV GF_PLUGIN_RENDERING_CHROME_BIN="/usr/bin/chrome"
# Disable Login form or not
ENV GF_AUTH_DISABLE_LOGIN_FORM "true"
# Allow anonymous authentication or not
ENV GF_AUTH_ANONYMOUS_ENABLED="true"
ENV GF_AUTH_ANONYMOUS_ORG_ROLE "Viewer"
# Role of anonymous user
ENV GF_AUTH_OAUTH_AUTO_LOGIN=false

USER root

#RUN rm -f /etc/ssl/certs/ca-certificates.crt
#COPY certs/45SLD/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

RUN mkdir -p "$GF_PATHS_PLUGINS" && \
  chown -R grafana:${GF_GID} "$GF_PATHS_PLUGINS" && \
  if [ $GF_INSTALL_IMAGE_RENDERER_PLUGIN = "true" ]; then \
  if grep -i -q alpine /etc/issue; then \
  apk add --no-cache udev ttf-opensans chromium && \
  ln -s /usr/bin/chromium-browser "$GF_PLUGIN_RENDERING_CHROME_BIN"; \
  else \
  cd /tmp && \
  curl -sLO https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb && \
  DEBIAN_FRONTEND=noninteractive && \
  apt-get update -q && \
  apt-get install -q -y ./google-chrome-stable_current_amd64.deb && \
  rm -rf /var/lib/apt/lists/* && \
  rm ./google-chrome-stable_current_amd64.deb && \
  ln -s /usr/bin/google-chrome "$GF_PLUGIN_RENDERING_CHROME_BIN"; \
  fi \
  fi

USER grafana

RUN if [ $GF_INSTALL_IMAGE_RENDERER_PLUGIN = "true" ]; then \
  grafana-cli \
  --pluginsDir "$GF_PATHS_PLUGINS" \
  --pluginUrl https://github.com/grafana/grafana-image-renderer/releases/latest/download/plugin-linux-x64-glibc-no-chromium.zip \
  plugins install grafana-image-renderer; \
  fi

ARG GF_INSTALL_PLUGINS="yesoreyeram-infinity-datasource,ae3e-plotly-panel,gapit-htmlgraphics-panel"

RUN if [ ! -z "${GF_INSTALL_PLUGINS}" ]; then \
  OLDIFS=$IFS; \
  IFS=','; \
  for plugin in ${GF_INSTALL_PLUGINS}; do \
  IFS=$OLDIFS; \
  if expr match "$plugin" '.*\;.*'; then \
  pluginUrl=$(echo "$plugin" | cut -d';' -f 1); \
  pluginInstallFolder=$(echo "$plugin" | cut -d';' -f 2); \
  grafana-cli --pluginUrl ${pluginUrl} --pluginsDir "${GF_PATHS_PLUGINS}" plugins install "${pluginInstallFolder}"; \
  else \
  grafana-cli --pluginsDir "${GF_PATHS_PLUGINS}" plugins install ${plugin}; \
  fi \
  done \
  fi

# use these if running docker-compose locally
COPY provisioning /etc/grafana/provisioning
COPY grafana.ini /etc/grafana/grafana.ini
COPY --from=conditional-copy /dashboard_subset/dashboards /etc/grafana/dashboards

```

### What did you expect to happen?

Dasbhoards still load if you are an anonymous user, and not require a login.

### Did this work before?

Yes it worked in v10.1.5

### How do we reproduce it?

Use the dockerfile and grafana.ini attached

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: 10.2.0
OS: Windows or Linux
Browser: Chrome

### Grafana platform?

Docker

**Comment by DeathRabbit679** [comment]
Got bit by this too using the public grafana 10.2.0 docker image. It looks like viewers cannot list dashboards, though if they know the exact dashboard url like `https://grafanahost:3000/d/210H456k/foobar?orgId=1` they can still view the dashboard in question. Though pretty rough for a viewer that doesn't have a bunch of bookmarks at the ready.

**Comment by hterik** [comment]
Same issue here. It appears like rbac was (partially?) enabled by default on OSS version, even if release notes only mention it being applicable to cloud version. 
Some endpoints like `/api/access-control/status` return 404, while others like `/api/access-control/user/permissions` are returning data. There one can see that Editor/Viewer users does not have permission to list dashboards for example

**Comment by n8ores** [comment]
I believe the issue is most likely related to the new release of "[public dashboards](https://grafana.com/docs/grafana/latest/dashboards/dashboard-public/ )".

It appears it has broken the previously implemented version for doing this, which was to set 

ENV GF_AUTH_ANONYMOUS_ORG_ROLE "Viewer"

**Comment by DeathRabbit679** [comment]
Could be related? https://github.com/grafana/grafana/issues/77121

**Comment by IevaVasiljeva** [comment]
Hi all! Thanks for reporting the issue!

I believe the issue is caused due to us failing to correctly set permissions on provisioned dashboards. I've got a fix for it here: https://github.com/grafana/grafana/pull/77155

A potential workaround until the fix is released would be for someone to log in as an Admin user and to manually set access to provisioned dashboards. This could either be done through the UI (using the dashboard and folder permissions tabs) or through API calls.

**Comment by pavel-mokrushin** [comment]
@IevaVasiljeva Is your fix supposed to hit only 10.3, not earlier?

**Comment by IevaVasiljeva** [comment]
@pavel-mokrushin currently yes. I will need to check internally if we're ok backporting it to 10.2.x.

**Comment by IevaVasiljeva** [comment]
This issue has been automatically closed as I've just merged the fix. I'll try to get this into the next 10.2.x patch, but meanwhile a workaround is to log in as an Admin user and manually set permissions on provisioned folders/dashboards to Viewers and Editors.

**Comment by IevaVasiljeva** [comment]
Looks like this fix will be a part of the next 10.2.x patch release (which should be 10.2.1) :)

**Comment by alelevinas** [comment]
> A potential workaround until the fix is released would be for someone to log in as an Admin user and to manually set access to provisioned dashboards. This could either be done through the UI (using the dashboard and folder permissions tabs) or through API calls.

So that ppl don't have to investigate on how to do this.
Get a token by [creating a service account](https://grafana.com/docs/grafana/latest/administration/service-accounts/#to-create-a-service-account) with an admin user

Get list of folders
`curl -v "<BASE_URL>/api/folders" -H "Authorization: Bearer <token>"`

[Post this](BASE_URL) to each folder uid
`curl -X POST -v <BASE_URL>:8080/api/folders/<folder_UID>/permissions -H "Authorization: Bearer <token>" -H 'Content-Type: application/json' -d '{"items": [{"role": "Viewer","permission": 1}]}'`

**Comment by Aaron-ML** [comment]
> Looks like this fix will be a part of the next 10.2.x patch release (which should be 10.2.1) :)

Any idea when 10.2.1 will be released? This is painful for our different environments right now.

**Comment by IevaVasiljeva** [comment]
> Any idea when 10.2.1 will be released?

It should be end of November. The current planned release date is 28th of November.

**Comment by n8ores** [comment]
As a temporary fix, you can also just continue to use 10.1.5. 

Thanks for fixing this.
